tests/topotests/bgp_flowspec/test_bgp_flowspec_topo.py

```diff
@@ -0,0 +1,178 @@
+
+"""
+
+import json
+import os
+import sys
+import pytest
+import getopt
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib import topotest
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+from lib.lutil import lUtil
+from lib.lutil import luCommand
+
+# Required to instantiate the topology builder class.
+from mininet.topo import Topo
+
+#####################################################
+##
+##   Network Topology Definition
+##
+#####################################################
+
+
+class BGPFLOWSPECTopo1(Topo):
+    "BGP EBGP Flowspec Topology 1"
+
+    def build(self, **_opts):
+        tgen = get_topogen(self)
+
+        # Setup Routers
+        tgen.add_router("r1")
+
+        # Setup Control Path Switch 1. r1-eth0
+        switch = tgen.add_switch("s1")
+        switch.add_link(tgen.gears["r1"])
+
+        ## Add eBGP ExaBGP neighbors
+        peer_ip = "10.0.1.101"  ## peer
+        peer_route = "via 10.0.1.1"  ## router
+        peer = tgen.add_exabgp_peer("peer1", ip=peer_ip, defaultRoute=peer_route)
+        switch.add_link(peer)
+
+
+#####################################################
+##
+##   Tests starting
+##
+#####################################################
+
+
+def setup_module(module):
+    tgen = Topogen(BGPFLOWSPECTopo1, module.__name__)
+
+    tgen.start_topology()
+    # check for zebra capability
+    router = tgen.gears["r1"]
+
+    # Get r1 reference and run Daemons
+    logger.info("Launching BGP and ZEBRA on r1")
+    router = tgen.gears["r1"]
+    router.load_config(
+        TopoRouter.RD_ZEBRA, os.path.join(CWD, "{}/zebra.conf".format("r1"))
+    )
+    router.load_config(
+        TopoRouter.RD_BGP, os.path.join(CWD, "{}/bgpd.conf".format("r1"))
+    )
+    router.start()
+
+    peer_list = tgen.exabgp_peers()
+    for pname, peer in peer_list.iteritems():
+        peer_dir = os.path.join(CWD, pname)
+        env_file = os.path.join(CWD, "exabgp.env")
+        peer.start(peer_dir, env_file)
+        logger.info(pname)
+
+
+def teardown_module(module):
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+
+def test_bgp_convergence():
+    "Test for BGP topology convergence"
+    tgen = get_topogen()
+
+    # Skip if previous fatal error condition is raised
+    topotest.sleep(10, "starting BGP peering with peer1")
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+
+def test_bgp_flowspec():
+    tgen = get_topogen()
+
+    # Skip if previous fatal error condition is raised
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    router = tgen.gears["r1"]
+
+    logger.info("Check BGP FS entry for 3.3.3.3 with redirect IP")
+    output = router.vtysh_cmd(
+        "show bgp ipv4 flowspec 3.3.3.3", isjson=False, daemon="bgpd"
```

Comment:
Where is the v6 test of data flow as well?

Reply:
I said during intiial meeting where we discussed with ipv6 that it was not possible to do ipv6 flowspec with exabgp.
so the test is limited to ipv4 flowspec.